Add support for OSD_EFFICIENCY

fiam opened the pull request:
Shows current efficiency as mah/km. Show also average efficiency
in the stats screen after disarming.

PR for configurator https://github.com/iNavFlight/inav-configurator/pull/274

fiam committed:
Add support for OSD_EFFICIENCY

Shows current efficiency as mah/km. Show also average efficiency
in the stats screen after disarming.

digitalentity commented on src/main/io/osd.c:
```diff
@@ -1405,6 +1405,27 @@ static bool osdDrawSingleElement(uint8_t item)
+            const int32_t maxEfficiencyValue = 999;
+            int32_t value = maxEfficiencyValue + 1;
+            if (gpsSol.groundSpeed > 0) {
+                value = (amperage / gpsSol.groundSpeed) / 0.0036f;
```
You need explicit conversion to `float` here. `amperage` and `gpsSol.groundSpeed` are integer. For efficiency below 278 mah/km `amperage / gpsSol.groundSpeed` will yield zero.

fiam replied:
I hadn't realised... what if we multiply `amperage` by `1/0.0036` (277.77, so we either pick 277 or 278) so we can remove the floating point math?

digitalentity replied:
Yeah, you can change to `value = 278 * amperage / gpsSol.groundSpeed;` Less than 0.1% error is marginal.

fiam replied:
Ok, let's do that

fiam committed:
Fix efficiency calculation and avoid FP math

fiam committed:
Add filtering to OSD_EFFICIENCY

fiam committed:
Don't draw OSD_EFFICIENCY if no GPS is configured

fiam committed:
Don't draw the efficiency if we have no GPS_FIX

Previously we were using the invalid GPS_SPEED acquired before
GPS_FIX was lost.

digitalentity commented on src/main/io/osd.c:
```diff
@@ -1628,6 +1670,16 @@ static void osdShowStats(void)
         strcat(buff, "\x07");
         displayWrite(osdDisplayPort, statValuesX, top++, buff);
+
+        if (STATE(GPS_FIX)) {
```
Should we check here? Isn't non-zero `totalDistance` a sign that we did have GPS at least some of the trip?

fiam committed:
Check for getTotalTravelDistance() > 0 to display efficiency in stats

If we lose GPS fix any time during the flight, the value is going
to lose accuraccy, so checking for distance travelled should be
enough and use a bit less of flash.
